app/controllers/api/v2/hosts_controller.rb

```diff
@@ -123,7 +123,7 @@ def destroy
       eos
 
       def status
-        render :json => { :status => @host.host_status }.to_json if @host
+        render :json => { :status => @host.get_status('HostStatus::ConfigurationStatus').to_label }.to_json if @host
```

Comment:
Indeed, status here should probably be either global or Puppet specific.

Adding actions won't break v2 compatibility, unless you're going to significantly change/remove existing actions.  If there was an API to retrieve different types of status including global and this existing route was deprecated, then it'd probably stay returning the Puppet status.